packages/use-intl/src/core/createFormatter.tsx

```diff
@@ -163,7 +163,7 @@ export default function createFormatter({
     value: Date | number,
     /** If a time zone is supplied, the `value` is converted to that time zone.
      * Otherwise the user time zone will be used. */
-    formatOrOptions?: string | DateTimeFormatOptions
+    formatOrOptions?: keyof IntlFormats['dateTime'] | DateTimeFormatOptions
```

Comment:
One important characteristic is that if the user doesn't supply types for global formats, then the overload of `DateTimeFormatOptions` still works. Seems to be the case based on my tests 👌

Reply:
But if the user doesn't supply the types for global formats, then the type should be as it was before yes? `string | DateTimeFormatOptions`

I'm still going to work on making it fallback to string when types are not provided

Reply:
If that is possible, that would be ideal—yes!